### PR TITLE
Add option to block startup until cluster join attempt has been made

### PR DESCRIPTION
**Pull request by howleysv**
This change will allow `libcluster_consul` to behave in line with the other strategies provided by `libcluster` (which all block startup to allow an initial connection) without changing behaviour for existing `libcluster_consul` users.

### CODE REVIEW

**Comment by howleysv** [comment]
@balena Is there anything I can do to help to get this merged?

**Comment by balena** [comment]
@szajbus I believe this may make sense. It doesn't conflict indeed.

File: README.md [hunk 1]
```diff
@@ -70,7 +70,16 @@ config :libcluster,
         # This is the node basename, the Name (first) part of an Erlang
         # node name (before the @ part. If not specified, it will assume
         # the same name as the current running node.
-        node_basename: "app_name"
+        node_basename: "app_name",
+
+        # This is the EEx template used to build the node names. The
+        # variables `ip`, `dc` and `node_basename` are available to
+        # compose the node name.
+        node_name_template: "<%= node_basename =>@<%= ip =>",
```
Comment (balena) [inline README.md]: This part (`node_name_template`) doesn't make sense anymore.

**Review by balena — CHANGES_REQUESTED** [review]
`node_name_template` does not exist anymore.

File: README.md (hunk 1 above)
Comment (howleysv) [reply README.md]: I removed this when resolving the merge conflict

**Review by szajbus — APPROVED** [review]
👍

**Pull request merged**